src/main/groovy/com/blackducksoftware/integration/hub/artifactory/inspect/blackDuckCacheInspector.groovy

```diff
@@ -35,96 +42,113 @@ import groovy.transform.Field
 @Field final boolean HUB_ALWAYS_TRUST_CERTS=false
 
 executions {
-    inspectRepository(httpMethod: 'GET', params:[:]) { params ->
-        if(!params['repos'][0] && !params['patterns'][0]) {
-            message = 'You must provide a comma separated list of repos and patterns'
-        } else {
-            Set<RepoPath> repoPaths = searchForRepoPaths(params['repos'][0], params['patterns'][0])
-            message = ''
-            for(RepoPath repoPath: repoPaths) {
-                populateHubOriginId(repoPath)
-            }
+    inspectRepository(httpMethod: 'POST') { params ->
+        if (!params['repoKey'][0] && !params['patterns'][0]) {
+            message = 'You must provide a repoKey and comma-separated list of patterns'
+            return
         }
+
+        createHubProject(params['repoKey'][0], params['patterns'][0])
     }
 
-    populateRepository(httpMethod: 'GET') { params ->
-        def repoName = params['repoName'][0]
+    updateInspectedRepository(httpMethod: 'POST') { params ->
+        def repoKey = params['repoKey'][0]
         def projectName = params['projectName'][0]
         def projectVersionName = params['projectVersionName'][0]
```

Comment:
We set this projectVersionName when we create the BDIO, why ask for it again? Couldn't we just get the hostname here as well?